pravega-sensor-collector/src/main/java/io/pravega/sensor/collector/file/FileProcessor.java

```diff
@@ -32,27 +32,27 @@
 
 /**
  * Get list of files obtained from config. Process each file for ingestion.
- * Keep track of new files and delete ingested files if "DELETE_COMPLETED_FILES"=true. 
+ * Keep track of new files and delete ingested files if "DELETE_COMPLETED_FILES"=true.
  */
-public class LogFileSequenceProcessor {
-    private static final Logger log = LoggerFactory.getLogger(LogFileSequenceProcessor.class);
+public abstract class FileProcessor {
+    private static final Logger log = LoggerFactory.getLogger(FileProcessor.class);
 
-    private final LogFileSequenceConfig config;
+    private final FileConfig config;
     private final TransactionStateSQLiteImpl state;
     private final EventWriter<byte[]> writer;
     private final TransactionCoordinator transactionCoordinator;
     private final EventGenerator eventGenerator;
 
-    public LogFileSequenceProcessor(LogFileSequenceConfig config, TransactionStateSQLiteImpl state, EventWriter<byte[]> writer, TransactionCoordinator transactionCoordinator, EventGenerator eventGenerator) {
+    public FileProcessor(FileConfig config, TransactionStateSQLiteImpl state, EventWriter<byte[]> writer, TransactionCoordinator transactionCoordinator) {
```

Comment:
TransactionStateSQLiteImpl -> This should be the interface

Reply:
done